userspace/libsinsp/sinsp_filtercheck_user.cpp

```diff
@@ -92,22 +92,22 @@ uint8_t* sinsp_filter_check_user::extract(sinsp_evt *evt, OUT uint32_t* len, boo
 	switch(m_field_id)
 	{
 	case TYPE_UID:
-		RETURN_EXTRACT_VAR(tinfo->m_user.uid);
+		RETURN_EXTRACT_VAR(tinfo->m_uid);
 	case TYPE_NAME:
-		RETURN_EXTRACT_CSTR(tinfo->m_user.name);
+		RETURN_EXTRACT_CSTR(tinfo->m_user->name);
 	case TYPE_HOMEDIR:
-		RETURN_EXTRACT_CSTR(tinfo->m_user.homedir);
+		RETURN_EXTRACT_CSTR(tinfo->m_user->homedir);
 	case TYPE_SHELL:
-		RETURN_EXTRACT_CSTR(tinfo->m_user.shell);
+		RETURN_EXTRACT_CSTR(tinfo->m_user->shell);
```

Comment:
Wouldn't this crash when dereferencing `m_user` that is nullptr (ie: when `SINSP_USER_DETAILS` is undefined)?